python/langsmith/evaluation/_arunner.py

```diff
@@ -635,6 +713,45 @@ async def aget_summary_scores(self) -> Dict[str, List[dict]]:
 
     ## Private methods
 
+    async def _apredict_and_evaluate(
+        self,
+        target: Optional[ATARGET_T],
+        evaluators: Sequence,
+        /,
+        max_concurrency: Optional[int] = None,
+    ) -> AsyncIterator[ExperimentResultRow]:
+        target_fn = _ensure_async_traceable(target) if target else None
+        evaluators = _resolve_evaluators(evaluators)
+
+        with cf.ThreadPoolExecutor(max_workers=4) as executor:
+
+            async def predict_all():
+                async for example in await self.aget_examples():
+                    # Yield the coroutine to be awaited later
+                    forward_result = await _aforward(
+                        target_fn,
+                        example,
+                        self.experiment_name,
+                        self._metadata,
+                        self.client,
+                        include_attachments=self._include_attachments,
+                    )
+                    current_results = {
+                        **forward_result,
+                        "evaluation_results": {"results": []},
+                    }
+                    experiment_row = self._arun_evaluators(
+                        evaluators, current_results, executor
+                    )
+                    yield experiment_row
+
+            async for result in aitertools.aiter_with_concurrency(
+                max_concurrency, predict_all(), _eager_consumption_timeout=0.001
```

Comment:
Is there a way to push this concurrency down to the predict_all layer, so we grab an example, then concurrently run over each example? Does the `aforward` and `arun_evaluators` get run sequentially per thread?